### PR TITLE
Update doc for the constraints of FractionalMaxPool2d

### LINKED ISSUE #115531: Hidden rule in nn.FractionalMaxPool2d
#### Description:

While using torch.nn.FractionalMaxPool2d, there is a hidden rule(seems to be
"kernel_size should be less than half of width and height of input tensor").
But this rule is not mentioned in the doc of FractionalMaxPool2d.

#### For example: 
```
x = torch.randn(3, 3, 7, 7)
p = torch.nn.FractionalMaxPool2d(kernel_size=4, output_ratio=0.8450560840300386, return_indices=False)
p(x)
```

this will tigger an error as below: 
```
----> 1 p(x)

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\pooling.py:799, in FractionalMaxPool2d.forward(self, input)
    798 def forward(self, input: Tensor):
--> 799     return F.fractional_max_pool2d(
    800         input, self.kernel_size, self.output_size, self.output_ratio,
    801         self.return_indices,
    802         _random_samples=self._random_samples)

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\_jit_internal.py:484, in boolean_dispatch.<locals>.fn(*args, **kwargs)
    482     return if_true(*args, **kwargs)
    483 else:
--> 484     return if_false(*args, **kwargs)

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\functional.py:503, in _fractional_max_pool2d(input, kernel_size, output_size, output_ratio, return_indices, _random_samples)
    492 if has_torch_function_variadic(input, _random_samples):
    493     return handle_torch_function(
    494         fractional_max_pool2d,
    495         (input, _random_samples),
   (...)
    501         _random_samples=_random_samples,
    502     )
--> 503 return fractional_max_pool2d_with_indices(
    504     input, kernel_size, output_size, output_ratio, return_indices, _random_samples
    505 )[0]

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\functional.py:482, in fractional_max_pool2d_with_indices(input, kernel_size, output_size, output_ratio, return_indices, _random_samples)
    480     n_batch = 1 if input.dim() == 3 else input.size(0)
    481     _random_samples = torch.rand(n_batch, input.size(-3), 2, dtype=input.dtype, device=input.device)
--> 482 return torch._C._nn.fractional_max_pool2d(input, kernel_size, output_size, _random_samples)

RuntimeError: fractional_max_pool2d(): pool height 4 too large relative to input height 7
```

### Suggest a potential alternative/fix

This should be supplemented in the documentation

cc @svekars @carljparker @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

### PR DESCRIPTION
**Pull request by huihoaan**
Fixes [#115531 ](https://github.com/pytorch/pytorch/issues/115531)
Update doc for the constraints of FractionalMaxPool2d.

### CODE REVIEW

**Comment by huihoaan** [comment]
@mikaylagawarecki cc

**Review by mikaylagawarecki — COMMENTED** [review]
Could we add a similar note in the documentation for `FractionalMaxPool3d` please? Note the similar checks [here](https://github.com/pytorch/pytorch/blob/61abacf8297837c13f58fbd99679085dc97fa1ab/aten/src/ATen/native/FractionalMaxPool3d.cpp#L73-L81)

File: torch/nn/modules/pooling.py [hunk 1]
```diff
@@ -735,6 +735,8 @@ class FractionalMaxPool2d(Module):
 
     .. note:: Exactly one of ``output_size`` or ``output_ratio`` must be defined.
 
+    .. note:: constraints of arguments: :math:`kH + oH -1 <= H_{in}` and :math:`kW + oW -1 <= W_{in}`
+
```
[the remark this answers is not in the document]
Comment (huihoaan) [reply torch/nn/modules/pooling.py]: I have moved this to `Args`,  `output_ratio` splited to `output_ratio_H` and `output_ratio_W`.

[the remark this answers is not in the document]
Comment (mikaylagawarecki) [reply torch/nn/modules/pooling.py]: Thank you! Could you please update FractionalMaxPool3d as well per the comment! :)

File: torch/nn/modules/pooling.py [hunk 2]
```diff
@@ -739,9 +739,12 @@ class FractionalMaxPool2d(Module):
         kernel_size: the size of the window to take a max over.
                      Can be a single number k (for a square kernel of k x k) or a tuple `(kh, kw)`
         output_size: the target output size of the image of the form `oH x oW`.
-                     Can be a tuple `(oH, oW)` or a single number oH for a square image `oH x oH`
+                     Can be a tuple `(oH, oW)` or a single number oH for a square image `oH x oH`.
+                     Note that we must have :math:`kH + oH -1 <= H_{in}` and :math:`kW + oW - 1 <= W_{in}`
```
Comment (mikaylagawarecki) [inline torch/nn/modules/pooling.py:743]: nit:
```suggestion
                     Note that we must have :math:`kH + oH - 1 <= H_{in}` and :math:`kW + oW - 1 <= W_{in}`
```

File: torch/nn/modules/pooling.py [hunk 3]
```diff
@@ -739,9 +739,12 @@ class FractionalMaxPool2d(Module):
...
         output_ratio: If one wants to have an output size as a ratio of the input size, this option can be given.
-                      This has to be a number or tuple in the range (0, 1)
+                      This has to be a number or tuple in the range (0, 1).
+                      Note that we must have :math:`kH + (output_ratio_H * H_{in}) - 1 <= H_{in}`
+                      and :math:`kW + (output_ratio_W * W_{in}) - 1 <= W_{in}`
```
Comment (mikaylagawarecki) [inline torch/nn/modules/pooling.py:747]: Oops, looks like the rendering is a bit off for this 
Screenshot 2023-12-27 at 12 37 39 PM

```suggestion
                      Note that we must have :math:`kH + (output\_ratio_H * H_{in}) - 1 <= H_{in}`
                      and :math:`kW + (output\_ratio\_W * W_{in}) - 1 <= W_{in}`
```

For future reference you can check the rendering in the [pytorchbot comment on the PR](https://github.com/pytorch/pytorch/pull/116261#issuecomment-1866075623) under 

> 📄 Preview [Python docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/116261/index.html)

**Review by mikaylagawarecki — APPROVED** [review]
Thanks!

File: torch/nn/modules/pooling.py [hunk 4]
```diff
@@ -739,9 +739,12 @@ class FractionalMaxPool2d(Module):
         kernel_size: the size of the window to take a max over.
                      Can be a single number k (for a square kernel of k x k) or a tuple `(kh, kw)`
         output_size: the target output size of the image of the form `oH x oW`.
-                     Can be a tuple `(oH, oW)` or a single number oH for a square image `oH x oH`
+                     Can be a tuple `(oH, oW)` or a single number oH for a square image `oH x oH`.
+                     Note that we must have :math:`kH + oH - 1 <= H_{in}` and :math:`kW + oW - 1 <= W_{in}`
         output_ratio: If one wants to have an output size as a ratio of the input size, this option can be given.
-                      This has to be a number or tuple in the range (0, 1)
+                      This has to be a number or tuple in the range (0, 1).
+                      Note that we must have :math:`kH + (output_\ratio\_H * H_{in}) - 1 <= H_{in}`
```
Comment (mikaylagawarecki) [inline torch/nn/modules/pooling.py:746]:
```suggestion
                      Note that we must have :math:`kH + (output\_ratio\_H * H_{in}) - 1 <= H_{in}`
```

**Pull request closed without merging**